include/DDCaloHitCreator.h

```diff
@@ -252,6 +268,12 @@ class DDCaloHitCreator
 
     dd4hep::VolumeManager m_volumeManager; ///< DD4hep volume manager
 
+    //Added by T.Pasquier
+    std::unique_ptr<pandora::ObjectFactory<object_creation::CaloHit::Parameters, object_creation::CaloHit::Object>> m_caloHitFactory{nullptr}; //General factory to initialize
+
+    /* DDCaloHitCreator& operator=(const DDCaloHitCreator&) = delete;  // Disallow copying
+    DDCaloHitCreator(const DDCaloHitCreator&) = delete; */
```

Comment:
```suggestion
public:
    DDCaloHitCreator& operator=(const DDCaloHitCreator&) = delete;
    DDCaloHitCreator(const DDCaloHitCreator&) = delete;
```
Since you know have a `unique_ptr` member copying won't work in any case. Marking these as `= delete` will improve the compiler error message. (I would move them to the rest of the constructors a bit further above).

Reply:
I uncommented it and moved it as suggested!